File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/types/non_void_function_definition.rb
# frozen_string_literal: true

module Seed
  module V2
    module V3
      module Problem
        module Types
          class NonVoidFunctionDefinition < Internal::Types::Model
            field :signature, lambda {
              Seed::V2::V3::Problem::Types::NonVoidFunctionSignature
            }, optional: false, nullable: false
            field :code, lambda {
              Seed::V2::V3::Problem::Types::FunctionImplementationForMultipleLanguages
            }, optional: false, nullable: false
          end
        end
      end
    end
  end
end

File: seed/php-sdk/trace/src/V2/Problem/Types/NonVoidFunctionDefinition.php
<?php

namespace Seed\V2\Problem\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class NonVoidFunctionDefinition extends JsonSerializableType
{
    /**
     * @var NonVoidFunctionSignature $signature
     */
    #[JsonProperty('signature')]
    public NonVoidFunctionSignature $signature;

    /**
     * @var FunctionImplementationForMultipleLanguages $code
     */
    #[JsonProperty('code')]
    public FunctionImplementationForMultipleLanguages $code;

    /**
     * @param array{
     *   signature: NonVoidFunctionSignature,
     *   code: FunctionImplementationForMultipleLanguages,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->signature = $values['signature'];$this->code = $values['code'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
